modules/gdscript/gdscript_compiler.cpp

```diff
@@ -213,7 +213,7 @@ static bool _have_exact_arguments(const MethodBind *p_method, const Vector<GDScr
 }
 
 GDScriptCodeGenerator::Address GDScriptCompiler::_parse_expression(CodeGen &codegen, Error &r_error, const GDScriptParser::ExpressionNode *p_expression, bool p_root, bool p_initializer, const GDScriptCodeGenerator::Address &p_index_addr) {
-	if (p_expression->is_constant) {
+	if (p_expression->is_constant && !p_expression->get_datatype().is_meta_type) {
```

Comment:
Actually, I just noticed that this is still in here and I'm getting this error in tests if it's not. It was part of why I created #70569, which was supposedly superseded by something else that didn't fix this particular issue (but fixed another).

Now I'm second guessing this, because it means it's not using the reduced_value dictionary.

```
modules\gdscript\tests\gdscript_test_runner.cpp(209): ERROR: CHECK( result.passed ) is NOT correct!
  values: CHECK( false )
  logged: C:/dev/godot/modules/gdscript/tests/scripts/analyzer/features/enum_function_typecheck.gd
          GDTEST_RUNTIME_ERROR
>> SCRIPT ERROR
>> on function: EnumFunctionTypecheckOuterClass.test()
>> analyzer/features/enum_function_typecheck.gd
>> 98
>> Invalid call. Nonexistent function 'inner_inner_no_class' in base 'RefCounted (InnerClass)'.
```

Reply:
I have not run your branch. But I would say it is using enum's reduced_value, just has to do a lookup instead of getting it immediately.

The issue, my bet, is not with enums, it is with classes. You have there a combo of a global class plus a nested class. This might be the thing with getting full script for the class.

Reply:
I was mulling it over while running chores and reached the same conclusion. I feel like it's ok to leave that in for now and look for a fuller solution later!